### PR TITLE
fix: Select align style broken with custom height

### LINKED ISSUE #44672: Cascader 组件下拉图标位置不正确
### Reproduction link

[https://stackblitz.com/edit/react-d9z3df?file=demo.js](https://stackblitz.com/edit/react-d9z3df?file=demo.js)

### Steps to reproduce

Cascader外容器有高度之后拉下图标位置不正确.

### What is expected?

外容器有高度后,下拉图标显示正确

### What is actually happening?

Cascader外容器有高度之后拉下图标位置不正确.

| Environment | Info |
| --- | --- |
| antd | 5.8.6 |
| React | 18.2 |
| System | window |
| Browser | chrome |

### PR DESCRIPTION
**Pull request by afc163**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

close https://github.com/ant-design/ant-design/issues/44672

### 💡 Background and solution

```
<Select style={{ height: 60 }} />
```

这样便能重现。

图片

修复后效果是：

图片

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  Fix Select/Cascader/TreeSelect style issue when customize their height.         |
| 🇨🇳 Chinese |  修复 Select/Cascader/TreeSelect 自定义高度时的样式问题。         |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at e1e6636</samp>

Improved the select component's demo and styles. Added more examples of option label centering to the demo using the `Cascader` component. Fixed some style issues in the `multiple.tsx` and `single.tsx` files to enhance the layout and alignment of the select component.

### <samp>🤖 Generated by Copilot at e1e6636</samp>

*  Add option label centering feature to different types of select components ([link](https://github.com/ant-design/ant-design/pull/44753/files?diff=unified&w=0#diff-05de49c151214d9ed1911d5c74659093c2f82e4c3be358d8b005d6a0b1c6acefL2-R9), [link](https://github.com/ant-design/ant-design/pull/44753/files?diff=unified&w=0#diff-05de49c151214d9ed1911d5c74659093c2f82e4c3be358d8b005d6a0b1c6acefL10-R39))
*  Adjust the height and vertical alignment of multiple select component to match the container and the other components ([link](https://github.com/ant-design/ant-design/pull/44753/files?diff=unified&w=0#diff-bdbff67724f9cb38a989e08057fac4b1dbbffcb715ecb8597db624ffd72f3ab5R57), [link](https://github.com/ant-design/ant-design/pull/44753/files?diff=unified&w=0#diff-bdbff67724f9cb38a989e08057fac4b1dbbffcb715ecb8597db624ffd72f3ab5R92), [link](https://github.com/ant-design/ant-design/pull/44753/files?diff=unified&w=0#diff-bdbff67724f9cb38a989e08057fac4b1dbbffcb715ecb8597db624ffd72f3ab5L184-R186))
*  Adjust the height and vertical alignment of single select component to match the container and the other components ([link](https://github.com/ant-design/ant-design/pull/44753/files?diff=unified&w=0#diff-d074c411bd6636664ae405c3b7a1744f68b0d236d82b2adbef155746711ec8a2R18), [link](https://github.com/ant-design/ant-design/pull/44753/files?diff=unified&w=0#diff-d074c411bd6636664ae405c3b7a1744f68b0d236d82b2adbef155746711ec8a2R46), [link](https://github.com/ant-design/ant-design/pull/44753/files?diff=unified&w=0#diff-d074c411bd6636664ae405c3b7a1744f68b0d236d82b2adbef155746711ec8a2L95-R97))

### CODE REVIEW

File: components/select/style/single.tsx [hunk 1]
```diff
@@ -92,7 +89,7 @@ function genSizeStyle(token: SelectToken, suffix?: string): CSSObject {
       [`&:not(${componentCls}-customize-input)`]: {
         [`${componentCls}-selector`]: {
           width: '100%',
-          height: token.controlHeight,
```
Comment (MadCcc) [inline components/select/style/single.tsx:95]: 这个高度在其他地方有设置么

File: components/auto-complete/demo/uncertain-category.tsx [hunk 2]
```diff
@@ -48,11 +48,12 @@ const App: React.FC = () => {
 
   return (
     <AutoComplete
-      dropdownMatchSelectWidth={252}
+      popupMatchSelectWidth={252}
```
Comment (afc163) [inline components/auto-complete/demo/uncertain-category.tsx:51]: 顺手改掉。

File: components/select/style/single.tsx [hunk 3]
```diff
@@ -15,6 +15,7 @@ function genSizeStyle(token: SelectToken, suffix?: string): CSSObject {
   return {
     [`${componentCls}-single${suffixCls}`]: {
       fontSize: token.fontSize,
+      height: token.controlHeight,
```
Comment (afc163) [inline components/select/style/single.tsx:18]: 改到外层了，这样可以被 style.height 覆盖。

File: components/select/style/single.tsx (hunk 1 above)
Comment (afc163) [reply components/select/style/single.tsx:95]: 高度改到外层了，这样可以被 style.height 覆盖。

**Review by kiner-tang — APPROVED** [review]

**Pull request merged**